htdocs/core/lib/tax.lib.php

```diff
@@ -285,6 +285,11 @@ function tax_by_date($type, $db, $y, $q, $date_start, $date_end, $modetax, $dire
         $sql.= " p.rowid as pid, p.ref as pref, p.fk_product_type as ptype,";
         $sql.= " 0 as payment_id, 0 as payment_amount";
         $sql.= " FROM ".MAIN_DB_PREFIX.$invoicetable." as f,";
+        if(! empty($conf->global->MAIN_TAX_USE_BANK_VALUE)) {
```

Comment:
This looks strange to me. Goal of this part of code is to get vat calculated on the "invoice" date and not on payment date.
Also, the vat report were redesigned into develop/8.0 to better address the need to calculate on invoice or payment depenging on company fiscal status, so take a look if you can make this change into 8.0 because it would not be possible to merge it into develop so i prefer cancel this for 7.0.